pyrdp/ui/qt.py

```diff
@@ -174,26 +183,40 @@ def setMousePosition(self, x: int, y: int):
         self.mouseY = y
         self.update()
 
+    def scale(self, scale):
+        """
+        Rescale the current widget to a percentage of the height of the RDP session.
+        :param scale: Ex: 0.5 for 50% height and 50% width.
+        """
+        self.scaleValue = scale
+        self.resize(self.sessionWidth, self.sessionHeight)
+
+    def setScaleToWindow(self, status):
+        self.scaleToWindow = status > 0
+
     def resize(self, width: int, height: int):
         """
         Resize widget
         :param width: new width of the widget
         :param height: new height of the widget
         """
-        self._buffer = QImage(width, height, QImage.Format_RGB32)
+        self._buffer = self._buffer.scaled(width, height)
```

Comment:
You shouldn't resize the rendering buffer. This will mess with PyRDP's rendering pipeline. You need to keep the original buffer full size and only scale it when it's time to paint the widget. I don't think it's going to be a performance issue, but if it is, we might need to have a debounce if Qt spams repaint events. in practice we'll be minimum-bound by the framerate, though.

Reply:
This is already a case. You're confusing `resize` and `scale`. `resize` already existed and was used to update the buffer size according to the screen size **when we receive clientData**. _buffer is only truely resized once, when clientData is parsed.

Hovewer, it is true that this is currently confusing and made me realise that `scale()` doesnt have to call `resize()` anymore, so I removed the call from `scale()`. I also clarified the docstring so its purpose is clearer.